feat(docs): add flag for building markdown docs

lgierth opened the pull request:
Look what I made \o/

This is for ipfs/docs#43

dignifiedquire commented on cmds/docs.js:
```diff
@@ -11,6 +11,11 @@ module.exports = {
       describe: 'Publish to GitHub Pages',
       default: false
+    },
+    markdown: {
```
This flag should also be present on the `release` command, as it generates docs as well.

dignifiedquire commented on src/docs/build.js:
```diff
@@ -50,6 +50,14 @@ function writeDocs (output) {
+  const docsPath = utils.getPathToDocs()
+  return fs.ensureDir(docsPath)
+    .then(() => {
+      fs.writeFileSync(docsPath + '/index.md', output)
```
please use `path.join(docsPath, 'index.md')` to ensure the path is usable on all oses

lgierth commented:
Hope I addressed all your points -- my JS is a bit rusty :)

Renamed the flag to `-d` because it conflicted with `-f` on the release command.

I didn't figure out how to test `node cli.js release --docs -d html -d md` without letting it touch github, but on the docs command it works, so it should be fine I guess?

lgierth commented:
I'm gonna call the windows node8 failure unrelated: "Error: EPERM: operation not permitted, scandir 'C:\projects\aegir\node_modules\fsevents\node_modules'"
